### PR TITLE
Context Alert Filter

### PR DESCRIPTION
**Pull request by tlenaic**
I am developing the feature "context alert filter" for jenkins.

Write now i have :

   -  add the section on Jelly.
    - The xml file should be in the directory filters of our job jenkins
    - I can parse my xml file
    - call the api alertfilter and addAlertFilter .

i am running some test now and my question is ruleId in the methode addAlertFilter of the class AlertFilter corresponds to what ?

My question is how could i find the id of my scanner (sql injection,...) ?

is it the value of pluginId in my ZAP xml report

thanks

frank

### CODE REVIEW

**Comment by tlenaic** [comment]
i don't now, if my question have been seen on the other link.
i want to know, if  in the methode addAlertFilter of the class AlertFilter, the string ruleId corresponds to the value of pluginId in my ZAP xml report ?

AND  for now, i am in my algorithm
 - i parse the xml file
 - add filters to the context
 - finally i run the active scan

do i, have miss a step ? because there is no changes in the report. 
i dont know if the method active scan check for filter in my context.
( I WORK with a session in jenkins without alerts, i erase all of them and i make sure the session was saved without )

**Comment by tlenaic** [comment]
i finally implemented correctly the feature.
and i have add logged out indicator because i need it in a project.
It is important to know that, the plugin use the unrelease version of zap-api-java. i needed the class alertFilter.

**Comment by JordanGS** [comment]
@tlenaic Hi, thank you for the contribution. I'll be checking over the PR over the weekend as well as the failure status that jenkins detected and let you know.

**Comment by tlenaic** [comment]
for this PR, the failure status is because i was using the alertfilter function. who is not present in the version 1.0.0 of zap-api realese.
i am preparing a correct one with zap-api 1.1.1 logged outindicator context alert filter and post build  step -->Management threshold

**Comment by JordanGS** [comment]
@tlenaic Hold off on any changes, I'll merge PR#3 in before yours. THen you can just merge those changes into your request since they contain fixed API calls. Is that okay?

**Comment by tlenaic** [comment]
okay

**Comment by JordanGS** [comment]
@tlenaic waiting on an issue from jenkinsci in relation to the build server. We'll talk on Sunday/Monday. Cheers mate.

**Comment by JordanGS** [comment]
@tlenaic New version is out and merged into master and development. Using API 1.1.1, feel free to update.

**Comment by JordanGS** [comment]
@tlenaic Can you squash the commits and more importantly rebase. That way it's all one commit and one commit message.

I'll test your PR tonight and tomorrow after work. Thanks!

P.S. Still some merge conflicts?

File: src/main/java/org/jenkinsci/plugins/zap/ZAPBuilder.java [hunk 1]
```diff
@@ -33,6 +33,8 @@
 import java.util.ArrayList;
 import java.util.concurrent.TimeUnit;
 
+import hudson.model.*;
```
Comment (JordanGS) [inline src/main/java/org/jenkinsci/plugins/zap/ZAPBuilder.java:36]: No wildcard imports, the previous was intentional.

```
import hudson.model.AbstractBuild;
import hudson.model.AbstractProject;
import hudson.model.BuildListener;
import hudson.model.Computer;
import hudson.model.Node;
```

File: src/main/java/org/jenkinsci/plugins/zap/ZAPBuilder.java [hunk 2]
```diff
@@ -69,6 +66,7 @@
  * @author Thilina Madhusanka
  * @author Johann Ollivier-Lapeyre
  * @author Ludovic Roucoux
+ * @author Lenaic Tchokogoue
```
Comment (JordanGS) [inline src/main/java/org/jenkinsci/plugins/zap/ZAPBuilder.java:69]: Most recent contributor at the top of the list.

File: src/main/java/org/jenkinsci/plugins/zap/ZAPBuilder.java [hunk 3]
```diff
@@ -338,7 +337,8 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
             listener.error(ExceptionUtils.getStackTrace(e));
             return false;
         }
-        return res;
+        //return res;
```
Comment (JordanGS) [inline src/main/java/org/jenkinsci/plugins/zap/ZAPBuilder.java:340]: Excess comment.

**Review by JordanGS — CHANGES_REQUESTED** [review]

**Comment by thc202** [comment]
Closing, there's a newer pull request (#4).

**Pull request closed without merging**